app/controllers/admin/videos_controller.rb

```diff
@@ -0,0 +1,22 @@
+class Admin::VideosController < AdminsController
+  def new
+    @video = Video.new
+  end
+
+  def create
+    @video = Video.new(admin_video_params)
+    if @video.save
+      flash[:success] = "You have successfully added the video '#{@video.title}'."
+      redirect_to new_admin_video_path
+    else
+      flash[:error] = "Your attempt to add a video failed! Please check errors!"
```

Comment:
`flash.now[:danger]`. "danger" having replaced "error" in the new bootstrap and `.now` being necessary because you are rendering not redirecting.